### PR TITLE
8439 - Datagrid Always on Icons

### PR DESCRIPTION
**Pull request by tjamesallen15**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

This pull request will add setting `showEditorIcons` to always display icon without hover.

**Related github/jira issue (required)**:
Closes https://github.com/infor-design/enterprise/issues/8439

**Steps necessary to review your pull request (required)**:
- Pull this git repository, build, and run the app
- Go to http://localhost:4000/components/datagrid/test-icon-always-on.html
- Icon should always display

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Review by glenlieorillo — CHANGES_REQUESTED** [review]
Datepicker and timepicker hover state is cut-off/misaligned.

https://github.com/infor-design/enterprise/assets/102151352/5204723a-f502-46c3-bfe0-0d51156bdd7e

image

image

File: src/components/datagrid/datagrid.js [hunk 1]
```diff
@@ -150,6 +150,7 @@ const COMPONENT_NAME = 'datagrid';
  * @param {array|object} [settings.attributes=null] Add extra attributes like id's to the toast element. For example `attributes: { name: 'id', value: 'my-unique-id' }`
  * @param {boolean} [settings.dblClickApply=false] If true, needs to double click to trigger select row in datagrid.
  * @param {boolean} [settings.allowPasteFromExcel=false] If true will allow data copy/paste from excel
+ * @param {boolean} [settings.showIcons=false] If true will always show hoverable icons.
```
Comment (tmcconechy) [inline src/components/datagrid/datagrid.js:153]: Maybe showEditorIcons to be more specific

**Review by tmcconechy — COMMENTED** [review]
A bunch of little problems... Ranked:

1. The rows change size when i click them
2. The icon should show the hover state (it only does when i enter the cell)
3. The time picker is missing the arrow on the popup
4. The dropdown icon changes position when opened
5. The time picker icon changes position when opened (in some rowheights)
6. Probably need to test that on all 4 row heights

**Review by jbrcna — CHANGES_REQUESTED** [review]
no tooltip arrow
icon not centered when hovered
no hover color when mouse is placed on top. it will display when clicked and hovered
image

**Comment by tmcconechy** [comment]
Looking better @tjamesallen15 still:

1) Can we make the buttons show the hover state on hover, also wondering why the color is grey on hover and not azure like https://main-enterprise.demo.design.infor.com/components/button/example-index.html
2) In extra small the icons shuffle (minorly)
3) In small  the icons shuffle (minorly)
4) In medium mode some of the text is sitting too high
5) Normal mode looks 👍🏻

**Comment by tmcconechy** [comment]
@tjamesallen15 still see

1. Uncentered text and icons in medium
2. No hover state on the icon when its "on"

**Review by tmcconechy — APPROVED** [review]

**Comment by tmcconechy** [comment]
@infor-design/qa can you re-review?

**Review by ericangeles — APPROVED** [review]

**Pull request merged**